### PR TITLE
3.0 preview dev

### PR DESCRIPTION
**Pull request by e111077**
tests fail on ie; waiting for baseuri polyfill to be added back into the webcomponentsjs polyfill

I also removed our url polyfill.

related bug https://github.com/webcomponents/webcomponentsjs/issues/964

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,27 +6,16 @@ addons:
...
-  - >-
-    npm run format && git diff --exit-code || (echo -e '\n\033[31mERROR:\033[0m
-    Project is not formatted. Please run "npm run format".' && false)
 script:
-  - xvfb-run polymer test
-  - >-
-    if [ "${TRAVIS_PULL_REQUEST}" = "false" ]; then polymer test -s 'default';
-    fi
+  - xvfb-run polymer test --module-resolution=node --npm
+  - 'if [ "${TRAVIS_PULL_REQUEST}" = "false" ]; then polymer test --module-resolution=node --npm -s ''default''; fi'
```
Comment (azakus) [inline .travis.yml]: No need for the single quotes around this whole line.

Comment (e111077) [reply .travis.yml]: odd idk why they're even there

File: package.json [hunk 2]
```diff
@@ -1,19 +1,32 @@
...
-    "update-types": "bower install && gen-typescript-declarations --deleteExisting --outDir .",
-    "format": "webmat && npm run update-types"
+    "format": "webmat"
+  },
+  "version": "3.0.0-pre.20",
+  "author": "The Polymer Authors",
+  "dependencies": {
+    "@polymer/polymer": "^3.0.0"
   }
 }
```
[the remark this answers is not in the document]
Comment (e111077) [reply package.json]: not here because some people might just want the query params instead

**Comment by e111077** [comment]
Those imports were not added manually, you might want to add ?w=1 for the diff, but I think modulizer is indeed being a bit eager here.

File: manifest.json [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  "files": {
+    "iron-location.html": {
+      "convertedUrl": "iron-location.js",
+      "exports": {}
+    },
+    "iron-query-params.html": {
+      "convertedUrl": "iron-query-params.js",
+      "exports": {}
+    },
+    "index.html": {
```
[the remark this answers is not in the document]
Comment (bicknellr) [reply manifest.json]: My mistake, after talking with @justinfagnani and looking in the modulizer source, I think this should actually be `"index.html": null,`: https://github.com/Polymer/polymer-modulizer/blob/2860badac55e0a2e989592bacd917e298ea1bf69/src/conversion-manifest.ts#L30

**Comment by bicknellr** [comment]
The demo seems to be broken because [`iron-demo-helpers/url-bar.js` imports iron-location](https://github.com/PolymerElements/iron-demo-helpers/blob/031603d2aa4d7fe9e0787359b4c3fc98f4a8af28/url-bar.js#L35), so it tries to register it a second time when [the demo imports it from `../iron-location.js`](https://github.com/PolymerElements/iron-location/blob/931264838134d02bd64761fbef6cb4c50ae9c2be/demo/index.html#L89).

**Comment by bicknellr** [comment]
Oh, I can edit these. :)

File: test/integration.html [hunk 4]
```diff
@@ -43,14 +37,14 @@
...
-import '../iron-location.js';
-import './replace-state-patch.js';
-import { Polymer } from '@polymer/polymer/lib/legacy/polymer-fn.js';
-Polymer({is: 'integration-element'});
-</script>
+      import '@polymer/polymer/polymer-legacy.js';
+      import '@polymer/promise-polyfill/promise-polyfill.js';
+      import '../iron-query-params.js';
+      import '../iron-location.js';
+      import './replace-state-patch.js';
```
Comment (bicknellr) [inline test/integration.html]: Could you move the imports that are duplicated here and in the script below up into a script in the head?

File: README.md [hunk 5]
```diff
@@ -1,61 +1,148 @@
...
+        <iron-location
+            path="/social/profiles"
+            hash="profilePicture"
+            query="{{paramsString}}"
+            dwell-time="1000">
+        </iron-location>
+        <iron-query-params
+            id="queryParams"
+            params-string='{{paramString}}'
+            params-object='{"userId": "polymer", "display": "dark"}'>
```
Comment (bicknellr) [inline README.md]: Should this example (and the in-element one below) be setting both `params-string` and `params-object`? It seems like one will overwrite the other.

**Comment by aomarks** [comment]
I just pushed a commit to this branch that removes the @polymer/promise-polyfill dependency

**Comment by bicknellr** [comment]
Maybe we should add a note at the top of the demo that explains that it's not loading the local one, in case someone's trying to use the demo to edit the element?

**Comment by e111077** [comment]
I'd like to also take the time to test IE. IE was having troubles with the webcomponents js polyfill (but last I remembered the issue just randomly fixed itself 🤷‍♂️)

**Comment by bicknellr** [comment]
Could you make the text actually visible in the demo, rather than just a comment?

**Comment by e111077** [comment]
done!

**Comment by e111077** [comment]
done!

**Comment by bicknellr** [comment]
Ping on this comment: https://github.com/PolymerElements/iron-location/pull/119#discussion_r205610588
Looks good otherwise.

**Comment by bicknellr** [comment]
Actually, there seems to be a legit failure in Edge 14?

**Comment by e111077** [comment]
I added it to the top:

https://github.com/PolymerElements/iron-location/pull/119/commits/176da009d4beb80fc9113cfaaf8ce351c430dc21

**Comment by e111077** [comment]
that one is a flake. Time to mash the refresh button!

**Comment by bicknellr** [comment]
> that one is a flake

Any idea why?

The `params-string` vs. `params-object` comment is the one I was referring to. The note in the demo looks good to me.

File: README.md (hunk 5 above)
Comment (e111077) [reply README.md]: the reason why iron-location is such a mess of a file is that it tries to resolve all the initial cases. In this case delcarative syntax should have precedent over the url. So this works as intended

**Pull request merged**